src/commonMain/kotlin/fr/acinq/lightning/payment/OutgoingPaymentFailure.kt

```diff
@@ -1,29 +1,71 @@
+    data object NoAvailableChannels : FinalFailure() { override fun toString(): String = "payment could not be sent through existing channels, check individual failures for more details" }
+    data class InsufficientBalance(val available: MilliSatoshi) : FinalFailure() { override fun toString(): String = "not enough funds in wallet to afford payment (available = $available)" }
+    data object NoRouteToRecipient : FinalFailure() { override fun toString(): String = "unable to route payment to recipient" }
+    data object RecipientUnreachable : FinalFailure() { override fun toString(): String = "the recipient was offline or did not have enough liquidity to receive the payment" }
+    data object RetryExhausted: FinalFailure() { override fun toString(): String = "payment attempts exhausted without success" }
+    data object WalletRestarted: FinalFailure() { override fun toString(): String = "wallet restarted while a payment was ongoing" }
+    data object UnknownError : FinalFailure() { override fun toString(): String = "an unknown error occurred" }
+    // @formatter:on
+}
+
+/**
+ * A non-final failure of a payment part, after which the payment may be retried after a while or using a different channel.
+ * Applications should define their own localized message for each of these failures.
+ */
+sealed class PartFailure {
+    // @formatter:off
+    /** The payment is too small: try sending a larger amount. */
+    data object PaymentAmountTooSmall : PartFailure()
+    /** The user has sufficient balance, but the payment is too big: try sending a smaller amount. */
+    data object PaymentAmountTooBig : PartFailure()
+    /** The user doesn't have sufficient balance: try sending a smaller amount. */
+    data object NotEnoughFunds : PartFailure()
+    /** The payment must be retried with more fees to reach the recipient. */
+    data object NotEnoughFees : PartFailure()
+    /** The payment expiry specified by the recipient is too far away in the future. */
+    data object PaymentExpiryTooBig : PartFailure()
+    /** There are too many pending payments: wait for them to settle and retry. */
+    data object TooManyPendingPayments : PartFailure()
+    /** Payments are temporarily paused while a channel is splicing: the payment can be retried after the splice. */
+    data object ChannelIsSplicing : PartFailure()
+    /** The channel is closing: another channel should be created to send the payment. */
+    data object ChannelIsClosing : PartFailure()
+    /** A remote node had a temporary failure: the payment may succeed if retried. */
+    data object TemporaryRemoteFailure : PartFailure()
+    /** The payment amount could not be relayed to the recipient, most likely because they don't have enough inbound liquidity. */
+    data object RecipientLiquidityIssue : PartFailure()
+    /** The payment recipient is offline and could not accept the payment. */
+    data object RecipientIsOffline : PartFailure()
```

Comment:
This would remove `isRouteError()`

```suggestion
    sealed class RouteError : FinalFailure()
    /** A remote node had a temporary failure: the payment may succeed if retried. */
    data object TemporaryRemoteFailure : RouteError ()
    /** The payment amount could not be relayed to the recipient, most likely because they don't have enough inbound liquidity. */
    data object RecipientLiquidityIssue : RouteError ()
    /** The payment recipient is offline and could not accept the payment. */
    data object RecipientIsOffline : RouteError ()
```

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/634/commits/ea4eb74b9fe11277ccf93a28ebb249820b322374